compiler/crates/relay-compiler/src/remote_persister.rs

```diff
@@ -8,8 +8,22 @@
 use crate::{OperationPersister, PersistConfig};
 use async_trait::async_trait;
 use persist_query::{persist, PersistError};
+use tokio::sync::Semaphore;
 
-pub struct RemotePersister;
+pub struct RemotePersister {
+    semaphore: Semaphore,
+}
+
+impl RemotePersister {
+    pub fn new(mut permits: usize) -> Self {
```

Comment:
nit: can we make this immutable? We usually prefer not to mutate params, unless there is no other way.

Reply:
fixed